### PR TITLE
[ticket/13282] Use 0 as default for integer type columns in postgresql

### PR DESCRIPTION
**Pull request by marc1706**
Needs different patch for develop.

Ticket: https://tracker.phpbb.com/browse/PHPBB3-13282

### CODE REVIEW

**Comment by marc1706** [comment]
@PayBas please check if this also fixes the issue for you

File: phpBB/phpbb/db/tools.php [hunk 1]
```diff
@@ -1574,7 +1574,15 @@ function sql_prepare_column_data($table_name, $column_name, $column_data)
 				}
 				else
 				{
-					$default_val = "'" . $column_data[1] . "'";
+					// Integers need to have 0 instead of empty string as default
+					if (preg_match('/int/i', $column_type))
```
Comment (nickvergessen) [inline phpBB/phpbb/db/tools.php]: I prefer `strpos() === 0`

**Comment by PayBas** [comment]
Can confirm fix, nice job.

File: phpBB/phpbb/db/tools.php (hunk 1 above)
Comment (marc1706) [inline phpBB/phpbb/db/tools.php]: I thought we might also support TINT but as it turns out there is no such thing in postgresql. Right now this is case insensitive though. Changing it to strpos() will modify this. Again, this should be fine as we always use uppercase for the column type in postgres.
I'll change this when I add the tests later today.

**Comment by marc1706** [comment]
@nickvergessen Added functional tests for this

**Comment by nickvergessen** [comment]
Since you change the tools.php,
mind adding a test for that in https://github.com/phpbb/phpbb/blob/develop/tests/dbal/db_tools_test.php#L39 ?

**Comment by marc1706** [comment]
Sure, will have to prepare a patch for develop anyway. If this PR looks fine to you I'll just base a branch on this and develop.

**Pull request merged**